utils/contextutils/context.go

```diff
@@ -0,0 +1,65 @@
+// Package contextutils provides utility for adding and retrieving metadata to/from a context.
+package contextutils
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type contextKey string
+
+const (
+	// MetadataKey is the key used to access metadata from a context with metadata.
+	MetadataKey = contextKey("viam-metadata")
```

Comment:
the convention is to call this MetadataContextKey

Reply:
Ah thanks! Fixed